In [ ]:
import pandas as pd

In [ ]:
dt = pd.read_csv("dataset.csv", encoding="latin1")
dt.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 297 entries, 0 to 296
Data columns (total 5 columns):
 #   Column                   Non-Null Count  Dtype  
---  ------                   --------------  -----  
 0   Programa/Fondo           297 non-null    object 
 1   Clave                    297 non-null    int64  
 2   Nombre del proyecto      297 non-null    object 
 3   Nombre del beneficiario  297 non-null    object 
 4    Monto de apoyo          297 non-null    float64
dtypes: float64(1), int64(1), object(3)
memory usage: 11.7+ KB


In [ ]:
print(dt.columns)

Index(['Programa/Fondo', 'Clave', 'Nombre del proyecto',
       'Nombre del beneficiario', ' Monto de apoyo '],
      dtype='object')


In [ ]:
mean_values = dt.groupby('Programa/Fondo')[' Monto de apoyo ']
print("Agrupamos los programas, y obtuvimos el promedio de monto de apoyo de c/u")
print(mean_values.mean())

Agrupamos los programas, y obtuvimos el promedio de monto de apoyo de c/u
Programa/Fondo
APPS                                 1.941253e+05
Alberta-Jalisco                      3.247912e+06
Becas CONACYT                        4.022527e+05
CLUSTER 2018                         2.642237e+06
Capital Intelectual 2018-2019        5.666223e+05
Diplomado PROTALENTO                 3.480000e+05
DyD                                  5.898589e+04
Epicentro 2018                       1.500000e+06
FEM                                  2.998200e+06
FORDECYT 2018-6                      2.500000e+06
Fondo Newton                         4.996200e+06
Industrias Creativas \n2018          1.339732e+06
Industrias creativas                 8.013423e+05
Innovación en la niñez y juventud    1.500000e+06
PRODEPRO                             2.759347e+05
PROPIN 2018-2                        3.563118e+04
PROSOFTJAL                           1.208871e+06
PROTALENTO 2018                      1.400000e+05
Singularity

In [ ]:
mean_values.describe()

count          mean           std  \
Programa/Fondo                                                         
APPS                                 5.0  1.941253e+05  1.238000e+04   
Alberta-Jalisco                      4.0  3.247912e+06  3.660323e+05   
Becas CONACYT                        9.0  4.022527e+05  2.592690e+05   
CLUSTER 2018                         1.0  2.642237e+06           NaN   
Capital Intelectual 2018-2019        6.0  5.666223e+05  3.964858e+05   
Diplomado PROTALENTO                 1.0  3.480000e+05           NaN   
DyD                                 24.0  5.898589e+04  2.907709e+04   
Epicentro 2018                       1.0  1.500000e+06           NaN   
FEM                                  1.0  2.998200e+06           NaN   
FORDECYT 2018-6                      1.0  2.500000e+06           NaN   
Fondo Newton                         1.0  4.996200e+06           NaN   
Industrias Creativas \n2018         10.0  1.339732e+06  3.734033e+05   
Industrias creativas                 2.0  8.013423e+05  9.880513e+05   
Innovación en la niñez y juventud    1.0  1.500000e+06           NaN   
PRODEPRO                            18.0  2.759347e+05  1.443899e+05   
PROPIN 2018-2                      149.0  3.563118e+04  2.064621e+04   
PROSOFTJAL                          28.0  1.208871e+06  1.048617e+06   
PROTALENTO 2018                      3.0  1.400000e+05  3.464102e+04   
Singularity                         32.0  1.641297e+05  2.372049e+05   

                                          min           25%          50%  \
Programa/Fondo                                                             
APPS                                172005.03  1.986214e+05   200000.000   
Alberta-Jalisco                    2706648.00  3.189162e+06  3392500.000   
Becas CONACYT                        15561.15  3.154232e+05   322262.120   
CLUSTER 2018                       2642237.00  2.642237e+06  2642237.000   
Capital Intelectual 2018-2019       264000.00  3.500000e+05   384967.000   
Diplomado PROTALENTO                348000.00  3.480000e+05   348000.000   
DyD                                  10000.00  3.252375e+04    68505.000   
Epicentro 2018                     1500000.00  1.500000e+06  1500000.000   
FEM                                2998200.00  2.998200e+06  2998200.000   
FORDECYT 2018-6                    2500000.00  2.500000e+06  2500000.000   
Fondo Newton                       4996200.00  4.996200e+06  4996200.000   
Industrias Creativas \n2018         299866.00  1.420869e+06  1493438.920   
Industrias creativas                102684.51  4.520134e+05   801342.255   
Innovación en la niñez y juventud  1500000.00  1.500000e+06  1500000.000   
PRODEPRO                             98055.00  1.498635e+05   283859.500   
PROPIN 2018-2                         3594.52  1.216000e+04    31376.980   
PROSOFTJAL                          205107.00  4.264348e+05   850504.000   
PROTALENTO 2018                     120000.00  1.200000e+05   120000.000   
Singularity                          31450.00  3.145000e+04    62900.000   

                                            75%         max  
Programa/Fondo                                               
APPS                               2.000000e+05   200000.00  
Alberta-Jalisco                    3.451250e+06  3500000.00  
Becas CONACYT                      3.971250e+05   941806.40  
CLUSTER 2018                       2.642237e+06  2642237.00  
Capital Intelectual 2018-2019      6.299835e+05  1315800.00  
Diplomado PROTALENTO               3.480000e+05   348000.00  
DyD                                8.000000e+04   100000.00  
Epicentro 2018                     1.500000e+06  1500000.00  
FEM                                2.998200e+06  2998200.00  
FORDECYT 2018-6                    2.500000e+06  2500000.00  
Fondo Newton                       4.996200e+06  4996200.00  
Industrias Creativas \n2018        1.499696e+06  1500000.00  
Industrias creativas               1.150671e+06  1500000.

In [ ]:
print("Agrupamos los programas, y obtuvimos la mediana del monto de apoyo de c/u")
print(mean_values.median())

Agrupamos los programas, y obtuvimos la mediana del monto de apoyo de c/u
Programa/Fondo
APPS                                  200000.000
Alberta-Jalisco                      3392500.000
Becas CONACYT                         322262.120
CLUSTER 2018                         2642237.000
Capital Intelectual 2018-2019         384967.000
Diplomado PROTALENTO                  348000.000
DyD                                    68505.000
Epicentro 2018                       1500000.000
FEM                                  2998200.000
FORDECYT 2018-6                      2500000.000
Fondo Newton                         4996200.000
Industrias Creativas \n2018          1493438.920
Industrias creativas                  801342.255
Innovación en la niñez y juventud    1500000.000
PRODEPRO                              283859.500
PROPIN 2018-2                          31376.980
PROSOFTJAL                            850504.000
PROTALENTO 2018                       120000.000
Singularity                  

In [ ]:
print("Agrupamos los programas, y obtuvimos la desviación estándar del monto de apoyo de c/u")
print(mean_values.std())

Agrupamos los programas, y obtuvimos la desviación estándar del monto de apoyo de c/u
Programa/Fondo
APPS                                 1.238000e+04
Alberta-Jalisco                      3.660323e+05
Becas CONACYT                        2.592690e+05
CLUSTER 2018                                  NaN
Capital Intelectual 2018-2019        3.964858e+05
Diplomado PROTALENTO                          NaN
DyD                                  2.907709e+04
Epicentro 2018                                NaN
FEM                                           NaN
FORDECYT 2018-6                               NaN
Fondo Newton                                  NaN
Industrias Creativas \n2018          3.734033e+05
Industrias creativas                 9.880513e+05
Innovación en la niñez y juventud             NaN
PRODEPRO                             1.443899e+05
PROPIN 2018-2                        2.064621e+04
PROSOFTJAL                           1.048617e+06
PROTALENTO 2018                      3.464102e+04

In [ ]:
print("Programa de apoyo más proporcionado:")
print(dt['Programa/Fondo'].mode())

Programa de apoyo más proporcionado:
0    PROPIN 2018-2
Name: Programa/Fondo, dtype: object


In [ ]:
print("Beneficiario más repetido")
print(dt['Nombre del beneficiario'].mode())

Beneficiario más repetido
0    Brian Alejandro Franco Ceja
Name: Nombre del beneficiario, dtype: object


In [ ]:
max1 = max(dt[' Monto de apoyo '])
print("Mayor donación")
print(max1)

Mayor donación
4996200.0


In [ ]:
min1 = min(dt[' Monto de apoyo '])
print("Menor donación")
print(min1)

Menor donación
3594.52


Tras realizar el análisis de los beneficiarios de programas de apoyo de julio a diciembre de 2018, podemos obtener las siguientes conclusiones:

- El beneficiario más repetido fue Brian Alejandro Franco Ceja
- El programa con más becas fue PROPIN 2018-2
- Hay un total de 19 fondos
- En base en la desviación estándar se concluye que hay progrmas que se brindaron con el mismo monto siempre, es decir que no cambia el monto del apoyo brindado
- La mayor donación fue de $4996200.0
- La menor donación fue de $3594.52
